eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentInitiator.scala

```diff
@@ -261,8 +261,14 @@ object PaymentInitiator {
     def invoice: Invoice
     def recipientNodeId: PublicKey = invoice.nodeId
     def paymentHash: ByteVector32 = invoice.paymentHash
-    def finalExpiry(nodeParams: NodeParams): CltvExpiry = nodeParams.paymentFinalExpiry.computeFinalExpiry(nodeParams.currentBlockHeight, invoice.minFinalCltvExpiryDelta)
     // @formatter:on
+    def finalExpiry(nodeParams: NodeParams): CltvExpiry = {
+      val minFinalCltvExpiryDelta = invoice match {
+        case invoice: Bolt11Invoice => invoice.minFinalCltvExpiryDelta
+        case _: Bolt12Invoice => CltvExpiryDelta(0)
```

Comment:
I'm still concerned about this, something isn't right here. If we use an expiry delta of 0 here, the recipient will receive an HTLC that expires at the current block height. This creates a non-negligible risk that the next-to-last node will force-close because the HTLC is expired before the last node answers with fulfill or fail. We introduced a non-0 final cltv expiry delta for Bolt 11 explicitly to avoid that issue, we don't want to reintroduce it for Bolt 12?

Reply:
No, the recipient won't receive an HTLC that expires at the current block height because there is a `minFinalCltvExpiryDelta` included in the blinded route, there is no problem here.
There was a problem in the invoice generation code where I didn't include the `minFinalCltvExpiryDelta` (it must have gotten lost in some rebase) but I've now fixed it.

Reply:
Great, that's a good solution, that works for me! It does need a unit test to verify that we correctly add this expiry delta when generating routes (which ensures that we don't regress here in a future change or rebase).

Reply:
The  `MultiPartHandlerSpec` tests check that.